### PR TITLE
Supporting ORM 2.5

### PR DESCRIPTION
**Pull request by smoench**

### CODE REVIEW

**Comment by OskarStark** [comment]
The Indention fix of the composer.json should ne in a separate PR to make only the version Change part of this PR

**Comment by Rvanlaak** [comment]
:+1: if this could be merged before symfony `2.8` is released

File: composer.json [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
+        "Persistence",
+        "Database",
+        "Audit"
+    ],
     "description": "Audit for Doctrine Entities",
     "require": {
         "doctrine/dbal": "~2.5",
-        "doctrine/orm": "~2.2"
+        "doctrine/orm": "~2.2",
+        "doctrine/doctrine-bundle": "~1.2"
```
Comment (DavidBadura) [inline composer.json]: why is it required?

**Comment by rvanlaak** [comment]
ping @DavidBadura

File: composer.json (hunk 1 above)
Comment (rvanlaak) [inline composer.json]: think this can be `~1.4`

**Comment by Soullivaneuh** [comment]
@smoench What is missing for get this merged? Need some help?

Is this PR working actually?

**Comment by smoench** [comment]
@Soullivaneuh one test is failing https://travis-ci.org/simplethings/EntityAudit/jobs/88157908#L294 
the failure was reported in #31 but the test case seems working in master (orm 2.4)

I think LogRevisionsListener::onFlush() and LogRevisionsListener::postFlush() needs some more improvements.

**Comment by Soullivaneuh** [comment]
Problem is related to issue deletion: https://github.com/simplethings/EntityAudit/blob/26134f53b4725a7a14e4e6df365943d7bcc4c390/tests/SimpleThings/Tests/EntityAudit/IssueTest.php#L101

If I comment it, test passes.

**Comment by rvanlaak** [comment]
Shouldn't you refresh the entity before removing it?

```php
$this->em->refresh($reve);
```

Hope that solves it

**Comment by Soullivaneuh** [comment]
@rvanlaak Nope, this change nothing.

**Comment by smoench** [comment]
@rvanlaak @Soullivaneuh tests are running green. would you please so kind as to test this branch in your projects. thanks in advance!

**Comment by rvanlaak** [comment]
Great! Will do that before the end of the year
Op 29 dec. 2015 6:59 p.m. schreef "Simon Mönch" <notifications@github.com>:

File: phpunit.xml.dist [hunk 2]
```diff
@@ -23,5 +23,6 @@
         <var name="DOCTRINE_USER" value=""/>
         <var name="DOCTRINE_PASSWORD" value=""/>
         <var name="DOCTRINE_HOST" value=""/>
+        <!--<var name="DOCTRINE_PATH" value="./db.sqlite"/>-->
```
Comment (rvanlaak) [inline phpunit.xml.dist]: This can be removed?

**Comment by rvanlaak** [comment]
I've tested the branch, and all my tests still pass. Added two minor comments :+1: 

Would be great if you could introduce a `CHANGELOG.md` with the new features?

File: composer.json [hunk 3]
```diff
@@ -1,19 +1,38 @@
...
     "description": "Audit for Doctrine Entities",
     "require": {
+        "php": "^5.3.9",
         "doctrine/dbal": "~2.5",
-        "doctrine/orm": "~2.2,<2.5"
+        "doctrine/orm": "~2.4"
     },
     "require-dev": {
+        "doctrine/doctrine-bundle": "~1.4",
         "symfony/framework-bundle": "~2.3",
```
Comment (rvanlaak) [inline composer.json]: Can we make `~2.3|~3.0` of this? Or maybe even `~2.7|~3.0`?

File: phpunit.xml.dist (hunk 2 above)
Comment (smoench) [inline phpunit.xml.dist]: No, this is great for debugging ...

File: composer.json (hunk 3 above)
Comment (smoench) [inline composer.json]: I will create an PR in the next few days as some deprecated features are still be used.

Comment (smoench) [inline composer.json]: Btw this is only a `require-dev` but it wouldn't working with symfony `~3.0` as of #165

File: src/SimpleThings/EntityAudit/AuditConfiguration.php [hunk 4]
```diff
@@ -35,6 +37,23 @@ class AuditConfiguration
...
+    public function getTableName(ClassMetadataInfo $metadata)
+    {
+        $tableName = $metadata->getTableName();
+
+        //## Fix for doctrine/orm >= 2.5
+        if (method_exists($metadata, 'getSchemaName') && $metadata->getSchemaName()) {
+            $tableName = $metadata->getSchemaName() . '.' . $tableName;
+        }
+
+        return $this->getTablePrefix() . $tableName . $this->getTableSuffix();
```
Comment (tolry) [inline src/SimpleThings/EntityAudit/AuditConfiguration.php]: Doesn't this concatination need to be done before prefixing the schema name? Else you might end up having `$tablePrefix.$schemaName.$tableName.$tableSuffix`

File: composer.json [hunk 5]
```diff
@@ -1,19 +1,38 @@
 {
     "name": "simplethings/entity-audit-bundle",
-    "license": "New BSD",
-    "keywords": ["Persistence", "Database", "Audit"],
+    "license": "LGPL-2.1",
```
Comment (tolry) [inline composer.json]: Why the license change?

Comment (smoench) [inline composer.json]: The license file says it is `LGPL-2.1`

File: src/SimpleThings/EntityAudit/AuditConfiguration.php (hunk 4 above)
Comment (smoench) [inline src/SimpleThings/EntityAudit/AuditConfiguration.php]: Indeed but with ORM 2.4 you would have the same result.

Comment (smoench) [inline src/SimpleThings/EntityAudit/AuditConfiguration.php]: opened an issue #166

**Comment by OskarStark** [comment]
:+1:

**Pull request merged**